developer/src/kmcmplib/src/Compiler.cpp

```diff
@@ -3389,12 +3389,20 @@ KMX_DWORD ReadLine(FILE* fp_in , PKMX_WCHAR wstr, KMX_BOOL PreProcess)
 
   if (cur == fsize)
 
-  // S: Is replacing "\r\n" with "\n" here sufficient or do we need changes at other places as well when we skip "\r"?
+  // \r\n is still added here even though Linux doesn`t use \r.
+  // This is to ensure to still have a working windows-only-version
   u16ncat(str, u"\r\n", _countof(str));  // I3481     // Always a "\r\n" to the EOF, avoids funny bugs
-  // u16ncat(str, u"\n", _countof(str));  // I3481
 
   if (len == 0) return CERR_EndOfFile;
 
+  // neccessary to add this block for using on non-windows platforms (removes all \r for platforms that use \n instead of \r\n)
+  for (p = str, n = 0; n < len; n++, p++) {
+    if (*p == L'\r')
+      *p = L' ';
+  }
+
+  // \r is still left in this block even though Linux doesn`t use \r.
+  // This is to ensure to still have a working windows-only-version
```

Comment:
This shouldn't be necessary because we've just stripped out all `\r` above.

Reply:
true it is not necessary, but if  we should ever want to use \r\n again it is easier to still leave it in the function than including it again at the right position.